File: Laboratorio1/Programas/Matrices/main.cpp
#include <iostream>

using namespace std;


int main()
{
    int tam;
    cout<<"Tamaño: ";
    cin>>tam;

    int matriz1[tam][tam];
    int matriz2 [tam][tam];
    int resultado[tam][tam];

        for(int i =0;i<tam;i++){
            for(int j=0;j<tam;j++){
                matriz1[i][j] = 1;
                matriz2[i][j] = 2;
            }
        }


     for(int i=0; i<tam; ++i){
         for(int j=0; j<tam; ++j){
            resultado[i][j] = 0;
         }

     }

        for (int x=0; x < tam; x++) {
            for (int y=0; y < tam; y++) {
                for (int z=0; z<tam; z++) {
                   resultado [x][y] += matriz1[x][z]*matriz2[z][y];
                }
            }
        }

  for (int n=0;n<tam;n++){
            for (int w=0;w<tam;w++){
                cout<<resultado[n][w]<<" ";
            }
            cout<<endl;
        }

    return 0;
}

File: Laboratorio1/Programas/Matrices/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    static int[][] multiplicacion(int k){
        int[][] matriz1 = new int[k][k];
        int[][] matriz2 = new int[k][k];
        for(int i =0;i<k;i++){
            for(int j=0;j<k;j++){
                matriz1[i][j] = 1;
                matriz2[i][j] = 2;
            }
        }
        int[][] resultado = new int[k][k];
        for (int x=0; x < k; x++) {
            for (int y=0; y < k; y++) {
                for (int z=0; z<k; z++) {
                   resultado [x][y] += matriz1[x][z]*matriz2[z][y];
                }
            }
        }
        return resultado;
    }

    static void imprimir(int[][] a,int k){
        for (int i=0;i<k;i++){
            for (int j=0;j<k;j++){
                System.out.print(a[i][j]+" ");
            }
            System.out.println();
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader lectura = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Tamaño: ");
        int tama = Integer.parseInt(lectura.readLine());
        imprimir(multiplicacion(tama),tama);
    }
}
